pthread linking error in Docker Ubuntu 18.04

```
[515/1403] Linking CXX executable libraries/fc/test/crypto/test_cypher_suites
FAILED: libraries/fc/test/crypto/test_cypher_suites
: && /usr/bin/clang++  -Wall -Wno-invalid-partial-specialization -fcolor-diagnostics -DWEBSOCKETPP_STRICT_MASKING -DBOOST_ASIO_HAS_STD_CHRONO -O2 -g -DNDEBUG   libraries/fc/test/crypto/CMakeFiles/test_cypher_suites.dir/test_cypher_suites.cpp.o  -o libraries/fc/test/crypto/test_cypher_suites  libraries/fc/libfc.a /usr/local/lib/libboost_thread.a /usr/local/lib/libboost_date_time.a /usr/local/lib/libboost_filesystem.a /usr/local/lib/libboost_system.a /usr/local/lib/libboost_program_options.a /usr/local/lib/libboost_signals.a /usr/local/lib/libboost_serialization.a /usr/local/lib/libboost_chrono.a /usr/local/lib/libboost_unit_test_framework.a /usr/local/lib/libboost_context.a /usr/local/lib/libboost_locale.a /usr/local/lib/libboost_iostreams.a -lpthread -L/usr/local/lib /usr/lib/x86_64-linux-gnu/libssl.a /usr/lib/x86_64-linux-gnu/libcrypto.a /usr/lib/x86_64-linux-gnu/libz.so -ldl -lrt libraries/fc/secp256k1/libsecp256k1.a /usr/lib/x86_64-linux-gnu/libgmp.a && :
/usr/lib/x86_64-linux-gnu/libcrypto.a(threads_pthread.o): In function `fork_once_func':
(.text+0x16): undefined reference to `pthread_atfork'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -36,6 +36,7 @@ SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a;.so")
 IF(NOT APPLE)
  # Linux or other unix
  SET(rt_library rt )
+ SET(CMAKE_THREAD_PREFER_PTHREAD TRUE)
  SET(pthread_library pthread)
 ELSE(NOT APPLE)
  ADD_COMPILE_FLAG("-D_GNU_SOURCE")

